server/routes/mentions.js

```diff
@@ -0,0 +1,76 @@
+const router = require("express").Router();
+
+const requiresAuth = require("./middleware/requiresAuth");
+const { Mention, User } = require("../models");
+
+const callScraper = require("../scraper");
+
+// TODO test mentions in action
+
+router.get("/", requiresAuth, async (req, res) => {
+  const user = await User.findByPk(req.user.id);
+  // note that the getter method for n-m can accept options
+  // as if it were a regular query
+  const mentions = await user.getMentions();
+
+  res.json(mentions);
+});
+
+router.get("/:id", requiresAuth, async (req, res) => {
+  const mention = await Mention.findByPk(req.params.id);
+
+  res.json(mention);
+});
+
+router.post("/", requiresAuth, async (req, res) => {
+  // NOTE I think this is better served as just a test route in order to test the scraper
+  // This shouldn't be placed in the final PR as there will be another means to post new data
+  // into the database.
+  const user = await User.findByPk(req.user.id);
+
+  let mentions = callScraper(req.user.company, 100);
+
+  let result = [];
+  for (let m of mentions) {
+    let userMentions = await user.getMentions({
+      where: {
+        title: m.title,
+      },
+    });
+
+    // if this is new
+    if (!userMentions.length) {
+      [mention, isNew] = await Mention.findOrCreate({
+        where: {
+          title: m.title,
+        },
+        defaults: {
+          title: m.title,
+          platform: m.platform,
+          date: m.date,
+          content: m.content || "",
```

Comment:
The content will be "" if it is empty when obtained from scraper, so I think the `|| ""` could be omitted?

Reply:
It's meant to safeguard in instances where an empty string isn't sent in the returning data. I believe this was the case where the reddit scraper was returning `undefined` instead of an empty string when the content value wasn't provided

Reply:
This is trivial indeed. I was considering efficiency concern, but the effect of a `||` can be ignored for sure.
Anyway, I think in the scrapers, no `undefined` value is returned. Correct me if you found any. 
So, if we do want defensive programming, it would be better to check each of the 4 fields is `null`, otherwise, I would say we can just skip this `||` operator. It is your call.

Reply:
I believe that is possible I didn't pull in your latest reddit scraper at the time where it does not return undefined, I will double check to make sure